pandas/tests/indexing/test_at.py

```diff
@@ -47,6 +37,68 @@ def test_selection_methods_of_assigned_col():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "value_to_assign",
+    [
+        "a",
+        1,
+        1.0,
+        np.nan,
+        True,
+        ("a",),
+        ["a"],
+        datetime(2000, 1, 2, tzinfo=timezone.utc),
+    ],
+)
+def test_at_setitem_expansion__df(value_to_assign):
+    # GH33544
+    # GH30649
+    result = DataFrame({0: [1, 2]}, index=[3, 2])
+
+    result.at[1, 1] = value_to_assign
+    expected_result = DataFrame(
+        {0: [1, 2, np.nan], 1: [np.nan, np.nan, value_to_assign]}, index=[3, 2, 1]
+    )
+    if isinstance(value_to_assign, datetime):
+        expected_result = expected_result.astype({1: object})
+
+    tm.assert_frame_equal(result, expected_result)
+
+
+@pytest.mark.parametrize(
+    "value_to_assign",
+    [
+        "a",
+        1,
+        1.0,
+        np.nan,
+        True,
+        ("a",),
+        ["a"],
+        {"a": "b"},
+        datetime(2000, 1, 2, tzinfo=timezone.utc),
+        np.array([1]),
+        Series([1]),
+    ],
+)
+@pytest.mark.filterwarnings("ignore::FutureWarning")
+def test_at_setitem_expansion__series(value_to_assign):
+    # GH33544
+    # GH30649
+    result = Series([1, 2], index=[3, 2])
+
+    result.at[1] = value_to_assign
+    # TODO remove if statement when concatenating bool-dtype and
+    # numeric-dtype is cast to object
+    if isinstance(value_to_assign, bool):
+        result = result.astype(object)
+    expected_result = Series([1, 2, value_to_assign], index=[3, 2, 1])
+    if isinstance(value_to_assign, datetime):
```

Comment:
umm this is a bug if it needs casting

Reply:
Hmm, I don't remember exactly why I added it in the first place, but it is indeed working without the expected_result cast. The first one is needed though for the case where `True` is being assigned to the series - it gets converted to 1 automatically when Series has `int64` dtype in the first place. Is that actually expected?